CMakeLists.txt

```diff
@@ -60,6 +64,12 @@ set(LUAU_OPTIONS)
 if(MSVC)
     list(APPEND LUAU_OPTIONS /D_CRT_SECURE_NO_WARNINGS) # We need to use the portable CRT functions.
     list(APPEND LUAU_OPTIONS /MP) # Distribute single project compilation across multiple cores
+
+    if(LUAU_STATIC_CRT)
+        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+        set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
+        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MTd")
```

Comment:
`CMAKE_MSVC_RUNTIME_LIBRARY` controls the runtime flags https://cmake.org/cmake/help/git-stage/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html thus `CMAKE_CXX_FLAGS_RELEASE` is redundant and not needed. However it requires the following policy to be active `cmake_policy(SET CMP0091 NEW)` in order to have any effect.
Having both `CMAKE_MSVC_RUNTIME_LIBRARY` and extra flags via `CMAKE_CXX_FLAGS_RELEASE` will cause build warnings

Reply:
Additionally, `CMAKE_MSVC_RUNTIME_LIBRARY` is available only since CMake `3.15` however root `CMakeLists` requires only `3.0`, thus additional version checks would be required.
Please note, that without `CMP0091 ` policy that enables `CMAKE_MSVC_RUNTIME_LIBRARY` and in all versions prior to 3.15, CMake forcefully sets runtime flags. Thus in order to change them a special cleanup need to be performed. This [StackOverflow post](https://stackoverflow.com/a/61603312/888545) shows how it can be done

Reply:
In the interest of simplicity I'm hoping we can simply error if CMake version is less than 3.15 and this build configuration is requested.

Reply:
That will work too. Thus new code should look like
```cmake
    if(LUAU_STATIC_CRT)
        cmake_minimum_required(VERSION 3.15)
        cmake_policy(SET CMP0091 NEW)
        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
```